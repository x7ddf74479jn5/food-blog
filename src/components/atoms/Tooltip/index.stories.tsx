import type { ComponentMeta, ComponentStory } from "@storybook/react";

import Tooltip from "./index";

export default {
  title: "Atoms/buttons/Tooltip",
  component: Tooltip,
} as ComponentMeta<typeof Tooltip>;

const Template: ComponentStory<typeof Tooltip> = (args) => (
  <Tooltip {...args}>
    <button className="p-1 text-black dark:text-white bg-green-500 rounded">children</button>
  </Tooltip>
);

export const Default = Template.bind({});
Default.args = {
  label: "label",
};
