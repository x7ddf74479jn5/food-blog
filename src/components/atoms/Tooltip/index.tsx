type Props = {
  label: string;
  children: React.ReactNode;
};

const Tooltip: React.FC<Props> = ({ label, children }) => {
  const base =
    "inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap";
  const hoverTransition = "opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in";

  return (
    <div className="group relative inline-block cursor-pointer">
      {children}
      {/* triangle */}
      <div className={`${base} ${hoverTransition} -bottom-4 h-3 w-3 rotate-45`} />
      <div className={`${base} ${hoverTransition} leading-snug} -bottom-8 p-1 text-xs`}>{label}</div>
    </div>
  );
};

export default Tooltip;
