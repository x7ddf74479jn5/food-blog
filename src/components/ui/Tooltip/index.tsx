import { classNames } from "@/utils/css";

type Props = {
  label: string;
  children: React.ReactNode;
};

const Tooltip: React.FC<Props> = ({ children, label }) => {
  return (
    <div className="group relative inline-block cursor-pointer" role="tooltip">
      {children}
      {/* triangle */}
      <div
        className={classNames(
          "inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap",
          "opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in",
          "-bottom-4 h-3 w-3 rotate-45"
        )}
      />
      <div
        className={classNames(
          "inline-block absolute z-10 text-white bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap",
          "opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in",
          "-bottom-8 p-1 text-xs leading-snug"
        )}
      >
        {label}
      </div>
    </div>
  );
};

export default Tooltip;
