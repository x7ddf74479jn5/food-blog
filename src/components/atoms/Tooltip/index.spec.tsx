import renderer from "react-test-renderer";

import Tooltip from ".";

describe("components/atoms/Tooltip", () => {
  it("snapshot", () => {
    const tree = renderer.create(<Tooltip label="label">children</Tooltip>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
