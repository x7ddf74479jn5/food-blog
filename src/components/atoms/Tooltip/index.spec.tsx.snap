// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/Tooltip snapshot 1`] = `
<div
  className="group inline-block relative cursor-pointer"
>
  children
  <div
    className="inline-block absolute bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-4 h-3 w-3 rotate-45"
  />
  <div
    className="inline-block absolute bg-gray-500 left-1/2 rounded-sm transform -translate-x-1/2 whitespace-nowrap opacity-0 invisible group-hover:visible group-hover:opacity-100 transition ease-in -bottom-8 p-1 text-xs leading-snug}"
  >
    label
  </div>
</div>
`;
